### PR TITLE
Fix errors at `enhanceRollupError` in Vite

### LINKED ISSUE #221: [Bug] Custom errors thrown during build are being swallowed by Vite
# FALSE, SEE COMMENT BELOW

### Describe the bug

The following stories file has two stories with the same name _"Default"_. It should throw a nice error here because this is not valid. But it doesn't. `dev` just ignores it, and `build` crashes with a hard to parse Vite error.

```svelte
<script lang="ts" context="module">
    import { defineMeta } from '@storybook/addon-svelte-csf';

    const { Story } = defineMeta({
        title: 'Cart/VoucherForm',
        component: VoucherForm,
    });
</script>

<script lang="ts">
    import VoucherForm from '@components/Cart/VoucherForm.svelte';
    import { expect, within } from '@storybook/test';
    import type { ComponentProps } from 'svelte';

    import { allModes } from '$storybook/modes';
</script>

{#snippet template(args: ComponentProps<VoucherForm>)}
    <VoucherForm {...args} />
{/snippet}

<Story
    name="Default"
    children={template}
/>

<Story
    name="Default"
    children={template}
/>
```

### PR DESCRIPTION
**Pull request by JReinhold**
Fixes #221 

When an error is thrown during build, Vite attempts to modify the error message with `error.message = "..."`:
https://github.com/vitejs/vite/blob/91a1acb12058d7f8ea357b7564992936eed62bc7/packages/vite/src/node/build.ts#L642

Our custom Error class didn't support that, because we add a `get message()` to the class. Adding a `set message()` too, ensures Vite doesn't thrown an error during step, so it actually is able to modify the error and throw the correct one being thrown by us.

In #221 it changes the error from:

```
=> Failed to build the preview
TypeError: Cannot set property message of  which has only a getter
    at enhanceRollupError (file://./node_modules/vite/dist/node/chunks/dep-BWSbWtLw.js:65318:15)
    at build (file://./node_modules/vite/dist/node/chunks/dep-BWSbWtLw.js:65456:5)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async build (./node_modules/@storybook/builder-vite/dist/index.js:58:230)
    at async Promise.all (index 0)
    at async Module.build2 (./node_modules/@storybook/builder-vite/dist/index.js:58:5160)
    at async Promise.all (index 0)
    at async buildStaticStandalone (./node_modules/@storybook/core/dist/core-server/index.cjs:46960:3)
    at async withTelemetry (./node_modules/@storybook/core/dist/core-server/index.cjs:47080:12)
    at async build (./node_modules/@storybook/core/dist/cli/bin/index.cjs:2841:3)
```

to the correct:

```
x Build failed in 1.12s
=> Failed to build the preview
[storybook:addon-svelte-csf-plugin-post] Duplicate exportNames found between two '<Story />' definitions in stories file: file://./tests/stories/test/Conflicts.stories.svelte

First instance: <Story name="First" exportName="First" ... />
Second instance: <Story name="First" exportName="First" ... />

This can happen when 'exportName' is implicitly derived by 'name'.
Complex names will be simplified to a PascalCased, valid JavaScript variable name,
eg. 'Some story name!!' will be converted to 'SomeStoryName'.
You can fix this collision by providing a unique 'exportName' prop with <Story exportName="SomeUniqueExportName" ... />.

More info: https://github.com/storybookjs/addon-svelte-csf/blob/v4.1.2/ERRORS.md#SB_SVELTE_CSF_PARSER_ANALYSE_STORY_PLUGIN_ERROR

file: ./tests/stories/test/Conflicts.stories.svelte
SB_SVELTE_CSF_PARSER_ANALYSE_STORY_0006 (DuplicateStoryIdentifiersError): Duplicate exportNames found between two '<Story />' definitions in stories file: file://./tests/stories/test/Conflicts.stories.svelte

First instance: <Story name="First" exportName="First" ... />
Second instance: <Story name="First" exportName="First" ... />

This can happen when 'exportName' is implicitly derived by 'name'.
Complex names will be simplified to a PascalCased, valid JavaScript variable name,
eg. 'Some story name!!' will be converted to 'SomeStoryName'.
You can fix this collision by providing a unique 'exportName' prop with <Story exportName="SomeUniqueExportName" ... />.

More info: https://github.com/storybookjs/addon-svelte-csf/blob/v4.1.2/ERRORS.md#SB_SVELTE_CSF_PARSER_ANALYSE_STORY_0006

    at getStoriesIdentifiers (./dist/parser/analyse/story/attributes/identifiers.js:56:19)
    at createAppendix (./dist/compiler/post-transform/create-appendix.js:14:30)
    at transformStoriesCode (./dist/compiler/post-transform/index.js:45:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.transform (./dist/compiler/plugins.js:82:13)
    at async file://./node_modules/vite-plugin-inspect/dist/index.mjs:377:17
    at async transform (file://./node_modules/rollup/dist/es/shared/node-entry.js:18676:16)
    at async ModuleLoader.addModuleSource (file://./node_modules/rollup/dist/es/shared/node-entry.js:18892:36)
```

### CODE REVIEW

**Review by xeho91 — APPROVED** [review]
I appreciate the thorough explanation of what happened ❤️

**Comment by shilman** [comment]
:rocket: PR was released in [`v5.0.0-next.8`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v5.0.0-next.8) :rocket:
